crates/rspack_core/src/resolver/mod.rs

```diff
@@ -169,6 +170,104 @@ which tries to resolve these kind of requests in the current directory too.",
+
+    let parent_path = match fs::metadata(&normalized_path) {
+      Ok(metadata) => {
+        // if the path is not directory, we need to resolve the parent directory
+        if !metadata.is_dir() {
+          normalized_path.parent()
+        } else {
+          is_resolving_dir = true;
+          Some(normalized_path.borrow())
+        }
+      }
+      Err(_) => normalized_path.parent(),
+    };
+
+    if parent_path.is_none() || file_name.is_none() {
+      return None;
+    }
+
+    let file_name = file_name.unwrap();
+    let parent_path = parent_path.unwrap();
+
+    // read the files in the parent directory
+    let files = fs::read_dir(parent_path);
+    match files {
+      Ok(files) => {
+        let mut requested_names = vec![file_name
+          .to_str()
+          .map(|f| f.to_string())
+          .unwrap_or_default()];
+        if is_resolving_dir {
+          // The request maybe is like `./` or `./dir` to resolve the main file (e.g.: index) in directory
+          // So we need to check them.
+          let main_files = dep
+            .resolve_options
+            .as_deref()
+            .or(Some(&plugin_driver.options.resolve))
+            .and_then(|o| o.main_files.as_ref().map(|f| f.clone()))
+            .unwrap_or_else(|| Vec::new());
+
+          requested_names.extend(main_files);
+        }
+
+        let suggestions = files
+          .into_iter()
+          .filter_map(|file| {
+            file.ok().and_then(|file| {
+              file.path().file_stem().and_then(|file_stem| {
+                if requested_names.contains(&file_stem.to_string_lossy().to_string()) {
+                  let mut suggestion = file.path().relative(args.context.as_path());
+
+                  if !suggestion.to_string_lossy().starts_with(".") {
+                    suggestion = PathBuf::from(format!("./{}", suggestion.to_string_lossy()));
+                  }
+                  Some(suggestion)
+                } else {
+                  None
+                }
+              })
+            })
+          })
+          .collect::<Vec<_>>();
+
+        if suggestions.len() == 0 {
+          return None;
+        }
+
+        let mut hint: Vec<String> = vec![];
+        for suggestion in suggestions {
+          let suggestion_ext = suggestion
+            .extension()
+            .map(|e| e.to_string_lossy())
+            .unwrap_or_default();
+          let suggestion_path = suggestion.to_string_lossy();
+          let specifier = args.specifier;
+
+          hint.push(format!(
+          "Found the module '{suggestion_path}' exists, but it was not resolved because its extension doesn't in the `resolve.extensions` list. Here are some possible solutions:
+
+1. add the extension '.{suggestion_ext}' to `resolve.extensions` in your rspack configuration
+2. use '{suggestion_path}' instead of '{specifier}'
+3. add the file path '{suggestion_path}' to the `entry` in your rspack configuration
```

Comment:
This seems not applicable to most cases, 1,2 looks good enough.

Reply:
ok I removed it.